File: walletsdk/src/main/java/leaf/prod/walletsdk/model/market/MarketInterval.java
/**
 * Created with IntelliJ IDEA.
 * User: kenshin wangchen@loopring.org
 * Time: 2018-09-19 下午2:32
 * Cooperation: loopring.org 路印协议基金会
 */
package leaf.prod.walletsdk.model.market;

import com.google.gson.annotations.SerializedName;

public enum MarketInterval {

    @SerializedName("OHLC_INTERVAL_ONE_MINUTES")
    ONE_MINUTE("1M"),

    @SerializedName("OHLC_INTERVAL_FIVE_MINUTES")
    FIVE_MINUTES("5M"),

    @SerializedName("OHLC_INTERVAL_FIFTEEN_MINUTES")
    FIFTEEN_MINUTES("15M"),

    @SerializedName("OHLC_INTERVAL_THIRTY_MINUTES")
    THIRTY_MINUTES("30M"),

    @SerializedName("OHLC_INTERVAL_ONE_HOUR")
    ONE_HOUR("1H"),

    @SerializedName("OHLC_INTERVAL_TWO_HOURS")
    TWO_HOURS("2H"),

    @SerializedName("OHLC_INTERVAL_FOUR_HOURS")
    FOUR_HOURS("4H"),

    @SerializedName("OHLC_INTERVAL_TWELVE_HOURS")
    TWELVE_HOURS("12H"),

    @SerializedName("OHLC_INTERVAL_ONE_DAY")
    ONE_DAY("1D"),

    @SerializedName("OHLC_INTERVAL_THREE_DAYS")
    THREE_DAYS("3D"),

    @SerializedName("OHLC_INTERVAL_FIVE_DAYS")
    FIVE_DAYS("5D"),

    @SerializedName("OHLC_INTERVAL_ONE_WEEK")
    ONE_WEEK("1W");

    private String description;

    MarketInterval(String description) {
        this.description = description;
    }

    public static MarketInterval getInterval(String description) {
        MarketInterval result = MarketInterval.ONE_DAY;
        for (MarketInterval interval : MarketInterval.values()) {
            if (interval.getDescription().equalsIgnoreCase(description)) {
                result = interval;
                break;
            }
        }
        return result;
    }

    public static MarketInterval getByName(String name) {
        MarketInterval result = MarketInterval.ONE_DAY;
        for (MarketInterval interval : MarketInterval.values()) {
            if (interval.toString().equalsIgnoreCase(name)) {
                result = interval;
                break;
            }
        }
        return result;
    }

    public String getDescription() {
        return description;
    }

    @Override
    public String toString() {
        String result = "1H";
        switch (this) {
            case ONE_HOUR:
                result = "1H";
                break;
            case TWO_HOURS:
                result = "2H";
                break;
            case FOUR_HOURS:
                result = "4H";
                break;
            case ONE_DAY:
                result = "1D";
                break;
            case ONE_WEEK:
                result = "1W";
                break;
        }
        return result;
    }
}

File: loopr-ios/Models/Market/MarketInterval.swift
//
//  TrendInterval.swift
//  loopr-ios
//
//  Created by xiaoruby on 12/1/18.
//  Copyright © 2018 Loopring. All rights reserved.
//

import Foundation

enum MarketInterval: String, CustomStringConvertible {

    case one_minute = "OHLC_INTERVAL_ONE_MINUTES"
    case five_minutes = "OHLC_INTERVAL_FIVE_MINUTES"
    case fifteen_minutes = "OHLC_INTERVAL_FIFTEEN_MINUTES"
    case thirty_minutes = "OHLC_INTERVAL_THIRTY_MINUTES"
    case one_hour = "OHLC_INTERVAL_ONE_HOUR"
    case two_hours = "OHLC_INTERVAL_TWO_HOURS"
    case four_hours = "OHLC_INTERVAL_FOUR_HOURS"
    case twelve_hours = "OHLC_INTERVAL_TWELVE_HOURS"
    case one_day = "OHLC_INTERVAL_ONE_DAY"
    case three_days = "OHLC_INTERVAL_THREE_DAYS"
    case five_days = "OHLC_INTERVAL_FIVE_DAYS"
    case one_week = "OHLC_INTERVAL_ONE_WEEK"

    var description: String {
        switch self {
        case .one_minute: return "1M"
        case .five_minutes: return "5M"
        case .fifteen_minutes: return "15M"
        case .thirty_minutes: return "30M"
        case .one_hour: return "1H"
        case .two_hours: return "2H"
        case .four_hours: return "4H"
        case .twelve_hours: return "12H"
        case .one_day: return "1D"
        case .three_days: return "3D"
        case .five_days: return "5D"
        case .one_week: return "1W"
        }
    }
}
